controllers/object_controls.go

```diff
@@ -1729,6 +1729,12 @@ func TransformDCGMExporter(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpe
 		setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), "DCGM_EXPORTER_COLLECTORS", MetricsConfigMountPath)
 	}
 
+	if len(config.DCGMExporter.Env) > 0 {
+		for _, env := range config.DCGMExporter.Env {
+			setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), env.Name, env.Value)
+		}
+	}
```

Comment:
Nit: the length check is redundant since a loop over an empty slice is always a no-op:

```suggestion
	for _, env := range config.DCGMExporter.Env {
		setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), env.Name, env.Value)
	}
```

Reply:
Updated as you have suggested.